File: src/jk_5/nailed/map/gameloop/instructions/InstructionSetFoodLevel.java
package jk_5.nailed.map.gameloop.instructions;

import jk_5.nailed.Nailed;
import jk_5.nailed.map.gameloop.GameThread;
import jk_5.nailed.map.gameloop.IInstruction;
import jk_5.nailed.players.Player;
import jk_5.nailed.teams.Team;

/**
 * No description given
 *
 * @author jk-5
 */
public class InstructionSetFoodLevel implements IInstruction {

    private String team;
    private int foodLevel;

    @Override
    public void injectArguments(String arguments) {
        String data[] = arguments.split(" ", 2);
        this.team = data[0];
        this.foodLevel = Integer.parseInt(data[1]);
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        Team team = controller.getMap().getTeamManager().getTeam(this.team);
        for (Player p : Nailed.playerRegistry.getPlayers()) {
            if (p.getEntity() == null) continue;
            if (p.getTeam() == team) {
                p.getEntity().getFoodStats().addStats(this.foodLevel - p.getEntity().getFoodStats().getFoodLevel(), 0);
            }
        }
    }
}

File: src/jk_5/nailed/map/gameloop/instructions/InstructionSetFoodLevel.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionSetFoodLevel extends IInstruction {
  private var team: String = null
  private var foodLevel: Int = 0

  def injectArguments(arguments: String) {
    val data = arguments.split(" ", 2)
    this.team = data(0)
    this.foodLevel = Integer.parseInt(data(1))
  }

  def execute(controller: GameThread) {
    val team = controller.getMap.getTeamManager.getTeam(this.team)
    if(team.isEmpty) return
    controller.getMap.getPlayers.filter(p => p.getTeam.eq(team.get) && p.getEntity.isDefined).map(_.getEntity.get.getFoodStats).foreach(s => s.addStats(this.foodLevel - s.getFoodLevel, 0))
  }
}
